src/rebar_deps.erl

```diff
@@ -457,8 +457,19 @@ is_app_available(Config, App, VsnRegex, Path, _IsRaw = false) ->
                               {AppFile, {expected, App}, {has, OtherApp}}}}}
             end;
         false ->
-            ?WARN("Expected ~s to be an app dir (containing ebin/*.app), "
-                  "but no .app found.\n", [Path]),
+            case filelib:is_dir(Path) of
+                true ->
+                    %% Path is a directory, but it's not an app dir.
+                    ?WARN("Directory expected to be an app dir, "
+                          "but no .app found:~n~s~n", [Path]);
```

Comment:
This error message makes it sound like the `.app` file should be at the root of the app dir, which is false. Sounds like keeping a mention of `ebin/*.app` would be useful.